stages/base_stage.go

```diff
@@ -22,18 +24,36 @@ import (
 	"github.com/recruit-tech/walter/log"
 )
 
+// BaseStage is an abstract struct implemented by the inherited struct that wishes to run somthing
+// in a stage.
 type BaseStage struct {
+	// Run specified process. The required method (Run) is defined in the inheirted structs.
 	Runner
-	InputCh     *chan Mediator
-	OutputCh    *chan Mediator
+
+	// Input channel.This channel stores the input of the statge.
```

Comment:
[nits]  It would be better to put space betwen `channel.` and `This` like this:

`Input channel.This` -> `Input channel. This`